formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -245,6 +247,40 @@ internal class FormulaManagerImpl<Input, State, Output>(
         return manager.run(input).output
     }
 
+    fun <ChildInput, ChildOutput> child(
+        key: Any,
+        formula: IFormula<ChildInput, ChildOutput>,
+        input: ChildInput,
+        onError: (Throwable) -> Unit,
+    ): ChildOutput? {
+        val childrenManager = getOrInitChildrenManager()
+        val manager = childrenManager.findOrInitChild(key, formula, input)
+
+        // If termination happens while running, we might still be initializing child formulas. To
+        // ensure correct behavior, we mark each requested child manager as terminated to avoid
+        // starting new actions.
+        if (isTerminated()) {
```

Comment:
I'm not sure how you would hit this

Reply:
I'm not entirely sure either, but it copied from the code here: https://github.com/instacart/formula/blob/15351c18b75be344622b9785649e92efb1778a15/formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt#L243

Reply:
Oh, I think it's if a side-effect terminates the root formula.